torch/csrc/jit/passes/constant_propagation.cpp

```diff
@@ -0,0 +1,126 @@
+#include "torch/csrc/jit/passes/constant_propagation.h"
+#include "torch/csrc/jit/passes/dead_code_elimination.h"
+#include "torch/csrc/utils/functional.h"
+#include "torch/csrc/jit/interpreter.h"
+#include "torch/csrc/jit/ivalue.h"
+#include "torch/csrc/autograd/variable.h"
+
+
+namespace torch { namespace jit {
+
+namespace {
+
+std::unordered_set<Symbol> skip_list = {
+  //FIXME If & Loop require special casing because they cannot be run as a
+  //single node.
+  prim::If,
+  prim::Loop,
+  //FIXME Same problem as in DCE - cpp & python PythonOp and CppOp should be
+  //FIXME treated as having side effects but ONNX depends on them being removed
```

Comment:
This is actually different in this case, because allowing them to run would remove them, so ONNX couldn't complain about that. You can't remove them in this pass exactly because they can have side effects that you don't want to trigger.